### ISSUE TITLE
Unbalanced quotes in line comments cause false `quote-danger` errors for Common Lisp

### ISSUE THREAD

**Issue by jimwhite** [eraserhd/parinfer-rust#156]
This is a problem I encountered trying to use parinfer on ACL2 (e.g. https://github.com/acl2/acl2/blob/master/axioms.lisp) which Copilot has fixed for me (PR to follow).
## Description

When processing Common Lisp files, parinfer-rust reports `Quotes must balanced inside comment blocks.` for lines where double-quote characters appear an odd number of times inside `;` line comments. This prevents processing of valid CL source files.

For example, in ACL2's `axioms.lisp` (a valid, certified Common Lisp file), comments like:

```lisp
; states, in "Constraints on the COMMON-LISP Package for Conforming
; Implementations," that: "conforming programs can use external symbols of the
; COMMON-LISP package..."
```

...trigger `quote-danger` because a quoted phrase wraps across line boundaries within a comment block.

## Why this is a false positive

In `;`-style line comments, the comment ends at the newline. A `"` inside a line comment is just prose — it cannot open a string that continues onto the next line. This is different from `#| ... |#` block comments where a quote could theoretically span multiple lines.

Parinfer currently toggles a `quote_danger` flag for every `"` in a comment and carries that state across newlines. This makes sense for block comments but is wrong for line comments.

## Reproduction

```bash
echo '; "unbalanced
(a b)' | parinfer-rust -l lisp -m paren
# => parinfer-rust: Quotes must balanced inside comment blocks.
```

This also affects `smart` and `indent` modes.

## Expected behavior

Line comments should not carry `quote_danger` state across newlines. Each `;` comment is self-contained.

## Environment

- parinfer-rust master (845bfbb)
- `-l lisp` mode
- Any mode (indent, paren, smart)

**Comment by eraserhd** [comment]
Unfortunately, this "quote danger" error is there to prevent Parinfer from corrupting valid files.  See [Inserting Quotes](https://shaunlebron.github.io/parinfer/#inserting-quotes) on the original Parinfer site.  It's specifically about what happens when unbalanced quotes appear in a single-line comment, which is syntactically valid in most? all? lisps, so yes, the Parinfer algorithm is explicitly preventing syntactically valid source code.  And I'm pretty sure this problem applies with Common Lisp.

A better solution might be to signal the error later - when inserting a quote prior to a single-line comment with unbalanced quotes.  e.g. "Cannot insert quote without corrupting file - single-line comments with odd numbers of quotes appear later in the file."

**Comment by eraserhd** [comment]
Another solution might be that in "paren" or "smart" mode, it will add the quotes, making the file safe to edit.  That seems consistent with how parinfer operates for indentation.

**Comment by jimwhite** [comment]
> Unfortunately, this "quote danger" error is there to prevent Parinfer from corrupting valid files. See [Inserting Quotes](https://shaunlebron.github.io/parinfer/#inserting-quotes) on the original Parinfer site. 

Is there a test for this?  I kinda get the idea but still whatever this feature is we need a test for it to ensure it works, "it" being the (non-)quoting/commenting behavior that is desired not the flag logic per se.  Then I (with Copilot's help) can look into how to get both things (avoid bad corrections and support ANSI CL).  

Changing the source code isn't an option.  I'm using parinfer to canonicalize the [ACL2](https://github.com/acl2/acl2) CL/ACL2 code in order to reason about it in a KG.  It turns out ACL2 is a bit of a torture test for ANSI CL spec compliance.  I [fixed one bug in tree-sitter-commonlisp](https://github.com/tree-sitter-grammars/tree-sitter-commonlisp/issues/48) that it turned up but when there were still more I switched to using rewrite-cl (https://github.com/jimwhite/pup/blob/main/external/script2notebook/lisp2nb.lisp).
